engine/src/main/java/org/terasology/config/flexible/FlexibleConfigImpl.java

```diff
@@ -32,24 +32,28 @@ public FlexibleConfigImpl() {
     }
 
     public <V> boolean add(Setting<V> setting) {
-        SimpleUri key = setting.getId();
+        SimpleUri id = setting.getId();
 
-        if (key == null || contains(key))
+        if (id == null) {
+            LOGGER.warn("The id of the setting can not be null.");
             return false;
+        } else if (contains(id)) {
+            LOGGER.warn("A Setting with the id \"{}\" already exists in the config.", id);
```

Comment:
"A Setting with the id \"{}\" already exists."

We can omit "in the config" because the logger already displays the name of the class from which the warning originates from.